Python/dfa.py

```diff
@@ -21,17 +21,30 @@ def __init__(self, alphabet):
         rejected: Flag to indicated whether a state ends up "dead" or there is not defined 
                   transisiton for the current input symbol
         curr_state : Keeps track of the state the DFA is currently in after each transistion 
+        __num_start_states : Number of states marked as start state in the DFA.
         """
         self.sigma = alphabet        
         self.transitions = {}
         self.states = {}
         self.rejected = 0
         
         self.curr_state = None
+        self.__num_start_states = 0
     
     def add_state(self, is_start, is_final, name):
         if name not in self.states:
             self.states[name] = State(is_start, is_final, name)
+
+            if is_start:
+                self.__num_start_states += 1
+            
+            if self.__num_start_states != 1:
```

Comment:
This condition won't work if the user decides to enter the start state after the addition of a first state. 

A better condition would be `self.__num_start_states > 1`

The same feedback goes for the c++ code as well.